### PR TITLE
Style

### PR DESCRIPTION
**Pull request by c-lliope**
[Work in Progress]

starting to restyle this page with inspiration from coreyhaines/GDCR

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -9,6 +9,7 @@
   <title>Coderetreat Live! - Coordination for simultaneous coderetreat events</title>
 
   <%= stylesheet_link_tag    "application" %>
+  <link href='http://fonts.googleapis.com/css?family=Amaranth:700|Roboto+Slab:400,700' rel='sty    lesheet' type='text/css'>
```
Comment (coreyhaines) [inline app/views/layouts/application.html.erb]: Is there extra spacing in the word stylesheet in the rel attribute? Looks like 

```
rel='sty lesheet'
```

**Comment by c-lliope** [comment]
Thanks @coreyhaines! Fixed it.

@jthurne and @coreyhaines I'm just about done styling the "running_today" page – check it out and see what you think. You'll see I had a bit of fun with it. (=

One problem I was running into is the rate limit on the Google Maps static image API – we might want to get an API key or cache the images on the server to prevent that. Without an API key, Google limits by IP address (https://developers.google.com/maps/documentation/staticmaps/index#Limits). Depending on how many events we have, if most people have their cache enabled while they refresh it might not be a problem.

I'm not sure if I'll have time to work on the admin pages, so you may want to work on those yourselves. I'll let you know if I have a chance to take a look at them.

**Comment by c-lliope** [comment]
> Depending on how many events we have, if most people have their cache enabled while they refresh it might not be a problem.

I guess I'd still feel more comfortable caching the images on the server.

**Comment by jthurne** [comment]
The Coderetreat Live! application has been known to get a LOT of traffic during the Global Day of Coderetreat, so your concern is very real.

**Comment by c-lliope** [comment]
Alright, I'm going to take a shot at the admin page.

**Comment by coreyhaines** [comment]
Yay! Thanks, so much. Do you want me to add you to the repo, so you can do
PRs from branches on the repo, rather than having to work from your own
fork?

**Comment by c-lliope** [comment]
Eh, it's not a big problem for me right now – any commits I push to my fork get included in this PR automatically. I don't think it'll speed up my workflow much.

If it's easier for you to check out branches on this repo, feel free to add me and I'll switch over.

**Comment by coreyhaines** [comment]
No worries. I usually merge them on github, so it isn't a big deal. I'll
add you probably tomorrow, though. The one speed up in the workflow is that
you can merge your own PRs. :)

**Comment by c-lliope** [comment]
Current Status:
screen shot 2014-11-08 at 6 56 46 pm

Would it be alright to require constraints when starting a new session? The cards with no constraints are a bit boring – they just display the time. Also, it doesn't look great on the home page (look at London):
screen shot 2014-11-08 at 7 00 46 pm

**Comment by coreyhaines** [comment]
That looks awesome.

Re: requiring constraints

I thought about this originally, but it has the potential to put an
unnecessary burden on the host/facilitator. I wanted to make sure that it
was as simple to use as possible. Every constraint means that the
host/facilitator has to think about something. If they aren't sure, or they
have a validation error, then they are less likely to use the system and
keep it updated.

**Comment by c-lliope** [comment]
Hrm... What if we autofill it with "Session 1", "Session 2", etc. if it's left blank?

**Comment by coreyhaines** [comment]
That could work. The key is to make it easy and low-stress for the
facilitators. If we auto-fill something there in order to make the UI look
nice, I'm cool with that.

**Comment by coreyhaines** [comment]
And this really looks pretty hot. I love it!

**Comment by jthurne** [comment]
Wow, Grayson you are a wizard! This is far more than I was expecting. The community is really going to love the new style.

**Comment by jthurne** [comment]
Since we are using images from Google Maps, do we need to include Google Map attribution ("Map data ©2014 Google") somewhere on the page?

**Comment by c-lliope** [comment]
Hm... I'm not sure. Here's the reference I was using: https://developers.google.com/maps/documentation/staticmaps/index

If you take a look at the maps, in the bottom-left corner they have the google logo on them. Since I'm cropping the images to circles, you can only see "..ogle"

I was going under the assumption that the watermark was all the attribution we needed – if someone pulls up the image in the devtools or opens it in a new tab, the cropping will be gone and they'll be able to see the image.

**Comment by c-lliope** [comment]
Alright! I think I'm about done.

I have one security concern, and if I could get another pair of eyes on it that would be great. I'm worried that a well-crafted URL entering [this controller action](https://github.com/coderetreat/coderetreat-live/pull/7/files#diff-68f16ac2d0eb8292d6ead535aa7d57b3R5) might give an attacker access to any file they want on the system.

Basically, the controller is taking a location from the URL (e.g. http://localhost:3000/map_images/San+Francisco,+CA ) and passing it into the MapImage class. The MapImage class tries to serve a file based on the location it is given. 

Any ideas on how to sanitize the input?

File: app/controllers/map_images_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require "map_image"
+
+class MapImagesController < ApplicationController
+  def show
+    send_file MapImage.for(params[:location]), disposition: :inline
```
Comment (c-lliope) [inline app/controllers/map_images_controller.rb:5]: Here's the possible security flaw

**Comment by jthurne** [comment]
Could you lookup the location in the DB before serving up the file? If the location isn't in the database, then don't try to fetch the file.

**Comment by jthurne** [comment]
You could also generate a list of map images that exist (list the directory) and then verify that the location passed in matches one of those file names (without the path).

**Comment by jthurne** [comment]
@graysonwright I just noticed that you did not check-in the public/map_images/default.png file. Could you check it in or pass it along to me? I'm finishing this up and getting ready to deploy it to production.

**Pull request merged**